### ISSUE TITLE
Synchronise with new features in pimoroni/pms5003-circuitpython 

### ISSUE THREAD

**Issue by kevinjwalters** [pimoroni/pms5003-python#14]
This library is very similar to pimoroni/pms5003-circuitpython and a lot of improvements were added to that in

- https://github.com/pimoroni/pms5003-circuitpython/pull/8
  - https://github.com/pimoroni/pms5003-circuitpython/issues/3
  - https://github.com/pimoroni/pms5003-circuitpython/issues/4
  - https://github.com/pimoroni/pms5003-circuitpython/issues/5
  - https://github.com/pimoroni/pms5003-circuitpython/issues/6

With some luck (and testing) the pms5003-python may be able to be generated from pms5003-circuitpython code with a minor patch. The versions could synchronise at `0.0.6` as a partial indication of their linkage.

**Comment by Gadgetoid** [comment]
Agreed. I just ported these to a new MicroPython library and they seem to work well!

I *think* it might be possible to abstract away the UART and GPIO access somewhat so that they can share a common codebase with just those portions changed.  There seem to be some platform-specific quirks regarding UART reads being blocking vs non-blocking and differing buffer sizes but hopefully nothing we can't smooth over.
